### PR TITLE
Fixed the issues with the tests in Eric's roi branch

### PR DESCRIPTION
**Pull request by sameera2004**

### CODE REVIEW

**Comment by sameera2004** [comment]
@ericdill tests are ok, issue with pandas
ERROR: Failure: ImportError (No module named pandas)

**Comment by sameera2004** [comment]
fixed the issue

**Comment by ericdill** [comment]
I am :-1: on including `mean_intensity_sets` as a library function, as it only iterates over a list and calls another library function `mean_intensity`.

**Comment by sameera2004** [comment]
@ericdill I took it out, now ready for review

File: .travis.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ install:
   - python setup.py install
   - pip install coveralls
   - pip install codecov
+  - pip install pandas
```
Comment (ericdill) [inline .travis.yml]: Can you `conda install pandas` instead? (Add it to line 18)

File: skxray/core/tests/test_roi.py [hunk 2]
```diff
@@ -278,49 +276,35 @@ def test_static_test_sets():
...
-    #  when trying to combine the values
-    assert_raises(ValueError,
-                  lambda: roi.combine_mean_intensity(average_int_sets,
-                                                     index_list2))
+    # get the mean intensities of image sets given as a dictionary
+    roi_data = {}
+    for k, v in sorted(samples.items()):
+        intensity, index_list = roi.mean_intensity(v, label_array)
+        roi_data[k] = intensity
+    average_int_sets = pd.DataFrame(roi_data, index_list)
```
Comment (ericdill) [inline skxray/core/tests/test_roi.py]: Can you rework this test to not use pandas?  I am not sure that it is a good idea to pick up a pandas dependency in the test suite because of this minor use.

**Comment by sameera2004** [comment]
@ericdill I took out the pandas dependency in these test, and rebased this PR. Ready for review.

File: skxray/core/roi.py [hunk 3]
```diff
@@ -357,139 +355,87 @@ def roi_pixel_values(image, labels, index=None):
...
-    ----------
-    images : array
-        Intensity array of the images
-        dimensions are: (num_img, num_rows, num_cols)
-
-    labels : array
-        labeled array; 0 is background.
-        Each ROI is represented by a distinct label (i.e., integer).
-
+    mean_intensity : list
```
Comment (tacaswell) [inline skxray/core/roi.py]: This is coming back an as array not a list

File: skxray/core/roi.py [hunk 4]
```diff
@@ -357,139 +355,87 @@ def roi_pixel_values(image, labels, index=None):
...
-    else:
-        raise ValueError("Labels list for the image sets are different")
-
-    return combine_mean_intensity
+        # use a mean that is mask-aware
+        mean_intensity[n] = ndim.mean(img, labeled_array, index=index)
+    # turn the 2-D array back into a list of arrays because the rows and
+    # columns have different units
+    data = [mean_intensity[:, i] for i in range(mean_intensity.shape[1])]
+    roi_labels = ['roi_%s' % idx for idx in index]
```
Comment (tacaswell) [inline skxray/core/roi.py]: What is the use for returning a list of strings here?  I think just returning `index` would be more useful here.

File: skxray/core/roi.py [hunk 5]
```diff
@@ -531,7 +473,6 @@ def roi_kymograph(images, labels, num):
     """
     roi_kymo = []
     for n, img in enumerate(images):
-        roi_kymo.append((roi_pixel_values(img,
-                                          labels == num)[0])[0])
+        roi_kymo.append((roi_pixel_values(img, labels == num)[0])[0])
```
Comment (tacaswell) [inline skxray/core/roi.py]: You can compute the label mask only once which will have a lot of time.

The two `[0]` makes is a bad code smell that makes it seem like there is something not quite right here.

File: skxray/core/roi.py (hunk 4 above)
Comment (ericdill) [inline skxray/core/roi.py]: I was matching the original API in my refactor of this code. I have no strong feelings either way

File: skxray/core/roi.py (hunk 5 above)
Comment (ericdill) [inline skxray/core/roi.py]: This could probably be done in some sort of horrible list comprehension too. Aren't those always faster?

Comment (ericdill) [inline skxray/core/roi.py]: The first zero is because `roi_pixel_values` returns two values: the list of pixels in each roi and the list of roi indices.  The first `[0]` is to grab the list of pixels in each roi and the second `[0]` is to grab the first and only element out of that list of pixels.  I agree that this is pretty ugly. How should that roi_pixel_values API be changed?

File: skxray/core/roi.py (hunk 4 above)
Comment (sameera2004) [inline skxray/core/roi.py]: Sure I will take this out(returning strings)

File: skxray/core/roi.py (hunk 3 above)
Comment (sameera2004) [inline skxray/core/roi.py]: Fixed the issue of mean_intensity coming as array, it returns a list

File: skxray/core/roi.py (hunk 5 above)
Comment (sameera2004) [inline skxray/core/roi.py]: took out one [0] out

**Comment by sameera2004** [comment]
Fixed the issues ready for review

File: skxray/core/roi.py (hunk 3 above)
Comment (tacaswell) [inline skxray/core/roi.py]: I would have just changed the documentation.  Arrays are far more useful than lists of lists

Comment (ericdill) [inline skxray/core/roi.py]: Yeah, I'm with Tom on this one. Can you undo the change to the code and update the docs to say that it is coming back as an array?

File: .travis.yml [hunk 6]
```diff
@@ -10,14 +10,17 @@ before_install:
   - chmod +x miniconda.sh
   - ./miniconda.sh -b -p /home/travis/mc
   - export PATH=/home/travis/mc/bin:$PATH
+  - wget https://gist.githubusercontent.com/tacaswell/128bb482f845feb024eb/raw/5cf21dc03a354fc87140d4a75e17cb5c076a0517/.condarc -O /home/travis/.condarc
```
Comment (ericdill) [inline .travis.yml]: why is this being added back in to the travis yaml?

File: skxray/core/roi.py [hunk 7]
```diff
@@ -357,139 +355,86 @@ def roi_pixel_values(image, labels, index=None):
...
-        shape (len(images), number of labels)
-
+        The labels for each element of the `mean_intensity` list
     """
-    if labels.shape != images[0].shape[0:]:
-        raise ValueError("Shape of the images should be equal to"
-                         " shape of the label array")
+    if labeled_array.shape != images[0].shape[0:]:
+        raise ValueError(
+            "`images` shape (%s) needs to be equal to the labeled_array` shape"
```
Comment (ericdill) [inline skxray/core/roi.py]: you have an extra backtick ( ` ) after `labeled_array`

File: skxray/core/roi.py [hunk 8]
```diff
@@ -531,7 +472,6 @@ def roi_kymograph(images, labels, num):
     """
     roi_kymo = []
     for n, img in enumerate(images):
-        roi_kymo.append((roi_pixel_values(img,
-                                          labels == num)[0])[0])
+        roi_kymo.append((roi_pixel_values(img, labels == num)[0]))
 
-    return np.matrix(roi_kymo)
+    return np.vstack(roi_kymo)
```
Comment (ericdill) [inline skxray/core/roi.py]: can you remove the `roi_` prefix from the `roi_kymo` local variable?

File: .travis.yml (hunk 6 above)
Comment (sameera2004) [inline .travis.yml]: It is a mistake I will take it out in travis yaml

File: skxray/core/roi.py (hunk 3 above)
Comment (sameera2004) [inline skxray/core/roi.py]: Sure I will make it to return an array instead of a list

**Comment by sameera2004** [comment]
Fixed ready for review

File: skxray/core/roi.py [hunk 9]
```diff
@@ -359,137 +357,83 @@ def roi_pixel_values(image, labels, index=None):
...
-    else:
-        raise ValueError("Labels list for the image sets are different")
-
-    return combine_mean_intensity
+        # use a mean that is mask-aware
+        data[n] = ndim.mean(img, labeled_array, index=index)
+    # turn the 2-D array back into a list of arrays because the rows and
+    # columns have different units
+    mean_intensity = [data[:, i] for i in range(data.shape[1])]
+    return np.asarray(mean_intensity), index
```
Comment (ericdill) [inline skxray/core/roi.py]: It is entirely unclear why you are laundering this numpy array through a list and back to a numpy array...

**Comment by sameera2004** [comment]
@ericdill now the mean_intensity only return as a bumpy array, took out the list

File: skxray/core/roi.py [hunk 10]
```diff
@@ -359,137 +357,80 @@ def roi_pixel_values(image, labels, index=None):
...
-
-    mean_intensity = np.zeros((images.shape[0], index.shape[0]))
+        index = list(np.unique(labeled_array))
+        index.remove(0)
+    try:
+        len(index)
+    except TypeError:
+        index = [index]
+    # not sure that this is needed
+    index = np.asarray(index)
```
Comment (ericdill) [inline skxray/core/roi.py]: This line is not needed. The index argument to `ndim.mean` just needs to be a sequence, not a numpy array

File: skxray/core/utils.py [hunk 11]
```diff
@@ -602,32 +602,30 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
...
-    Make a grid of radial positions.
+    """Convert a cartesian grid (x,y) to the radius relative to some center
 
     Parameters
     ----------
     center : tuple
         point in image where r=0; may be a float giving subpixel precision.
         Order is (rr, cc).
-
     shape: tuple
```
Comment (ericdill) [inline skxray/core/utils.py]: add a whitespace between `shape` and `:`

File: skxray/core/utils.py [hunk 12]
```diff
@@ -602,32 +602,30 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
...
-        pixel coordinates. Order is (rr, cc).
+        pixel coordinates.
+        Order is (rr, cc).
 
     Returns
     -------
     r : array
-        The L2 norm of the distance of each pixel from the calibrated center.
+        The distance of each pixel from `center`
+        Shape of the array is equal to `shape`
```
Comment (ericdill) [inline skxray/core/utils.py]: `Shape of the return value is equal to the `shape` input parameter

File: skxray/core/utils.py (hunk 11 above)
Comment (ericdill) [inline skxray/core/utils.py]: Also add 
```
pixel_size : sequence, optional
    The physical size of the pixels.
    len(pixel_size) should be the same as len(shape)
    defaults to (1,1)
```

**Comment by ericdill** [comment]
Great, thanks for all the work!

**Comment by sameera2004** [comment]
Sure, Thank you for all the help!
